pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,68 @@
+package agent
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Platform stores cluster configuration for an agent based install where hosts are booted and registered, ready
+// to be assigned and used in a cluster.
+type Platform struct {
```

Comment:
What is the bare minimum set of required params for assisted install? Most things are marked omitempty here, implying they are optional, but I suspect a few may actually be mandatory.

Reply:
In order to start installation we need most of the parameters, just to create a cluster you need name, openshift version and pull secret ref.

Reply:
Hmm interesting, ok so moving to declarative we should assume there's no distinction there anymore, and all of these properties are now mandatory. I will drop the omitempties.